src/Song.cpp

```diff
@@ -449,6 +449,15 @@ bool Song::LoadFromSongDir(RString sDir, bool load_autosave, ProfileSlot from_pr
 		}
 	}
 
+	// Apply the group offset to the song's timing data
+	Group* pGroup = SONGMAN->GetGroupByName(m_sGroupName);
+	if (pGroup != nullptr) {
+		m_SongTiming.m_fBeat0GroupOffsetInSeconds = pGroup->GetSyncOffset();

```

Comment:
This right here is the legitimate bug and I believe is the actual solution.

It's possible we weren't originally setting m_fBeat0GroupOffsetInSeconds in the cases where TidyUpData wasn't called, so the offset would be it's default value (which is 0.0f) instead of what it should have been.

Reply:
Handled in: https://github.com/itgmania/itgmania/pull/693

Wanted to make the next RC before I went to bed.